phc-rails/app/models/account.rb

```diff
@@ -41,7 +41,80 @@ def self.spawn(params)
     if account.save then account else nil end
 
     #### TODO: POST TO SALESFORCE #####
+  end
+
+  def self.find_duplicates_by(attrs, count, cursor)
+    groups = {}
+    # Escape each attribute with quotes to prevent lowercasing by Rails
+    attrs.map! { |x| "\"#{x}\"" }
+    result = Account.select(attrs).group(attrs).having("count(*)>1")
+    result[cursor..cursor+count].each do |account|
+      fields = account.attributes
+      fields.delete('id')
+      accounts = Account.where(fields)
+      groups[fields] = accounts.map(&:to_hash)
+    end
+    groups
+  end
+
+  def self.find_new_accounts()
+    # Returns a list of new accounts made at the last PHC event, aka ones with no Salesforce ID
+    Account.where(sf_id: nil)
+  end
+
+  def to_hash
+    {
+      name: name(self.FirstName, self.LastName),
+      ssn: ssn(self.SS_Num__c),
+      birthday: self.Birthdate__c,
+      sf_id: self.sf_id,
+      created_at: self.created_at,
+      phone: self.Phone
+    }
+  end
+
+  def name(first, last)
+    if first.nil? or first.empty?
+      first = "(None)"
+    end
+    if last.nil?
+      last = ""
+    end
+    "#{first} #{last}"
+  end
 
+  def ssn(ssn)
+    if ssn.nil? or ssn.empty?
+      "(None)"
+    elsif ssn.length < 9
+      ssn
+    else
+      "x"*6 + self.SS_Num__c[-4..-1]
+    end
   end
 
+  def birthdate()
```

Comment:
Assignment Branch Condition size for birthdate is too high. [22.56/15]<br>Omit the parentheses in defs when the method doesn't accept any arguments.